packages/spright-components/src/demo/template.ts

```diff
@@ -0,0 +1,22 @@
+import { html } from '@microsoft/fast-element';
+import { iconUserTag } from '@ni/nimble-components/dist/esm/icons/user';
+import type { Demo } from '.';
+
+export const template = html<Demo>`
+    <div
+        class="backplate ${x => x.shape}"
+        part="backplate"
+        style="${x => (x.fill ? `background-color: ${x.fill};` : undefined)}"
+    >
+        <a
+            class="link"
+            part="link"
+            href="${x => (x.link ? x.link : undefined)}"
+            style="${x => (x.color ? `color: var(--avatar-color-${x.color});` : undefined)}"
```

Comment:
Where is `--avatar-color-...` set?

Reply:
Chatted with Jesse and lets just simplify the component further and give it a name that is a clear responsibly.

spright-rectangle

template.ts
```
<slot></slot>
```

styles.ts
```
display inline-block

:host {
  // border color tokens
}

slot {
   // font tokens
}
```

Scope of "demo" is unclear and could see it growing unnecessarily to demonstrate lots of things like custom tokens, accessibility patterns, component composition, plugins, whatever. Instead spright-rectangle is super scoped and small.

Reply:
Also see the statement from the other comment about being derived from FAST Foundation but I think we just keep it even simpler and derive from FASTElement. I kinda expect long-term most things in FAST Foundation to be in Nimble instead of spright and even if extending from FAST Foundation nimble has plenty of examples already.

My understanding is nimble-rectangle is super simple and something just for infrastructure set-up that we can remove when we get a real spright component

Reply:
> Also see the statement from the other comment about being derived from FAST Foundation...

Not sure what you're referring to.

> ...but I think we just keep it even simpler and derive from FASTElement.

Needed to derive from `FoundationElement` in order to have a `compose` method to call.

Reply:
>> Also see the statement from the other comment about being derived from FAST Foundation...
> 
> Not sure what you're referring to.

That was my interpretation of this statement: https://github.com/ni/nimble/pull/1986#discussion_r1571370842

> I went with option 3, but I wanted something that derived from a FAST component, so I made spright-demo which is actually a not-very-functional user avatar.

That it was desired to extend a component that already existed in Fast Foundation (assumed that was the intention of "derived from a FAST component").

Reply:
Got it. 👍I thought you were directing me to refer to a comment _you_ made elsewhere, when you meant that you saw _my_ comment.